test/widgets/home_test.dart

```diff
@@ -168,6 +171,44 @@ void main () {
       check(find.byType(BottomSheet)).findsOne();
     }
 
+    /// Taps the [buttonFinder] button and awaits the bottom sheet's exit.
+    ///
+    /// Includes a check that the bottom sheet is gone.
+    /// Also awaits the transition to a new pushed route, if one is pushed.
+    ///
+    /// [buttonFinder] will be run only in the bottom sheet's subtree;
+    /// it doesn't need its own `find.descendant` logic.
+    Future<void> tapButtonAndAwaitTransition(WidgetTester tester, Finder buttonFinder) async {
+      final topRouteBeforePress = topRoute;
+      check(topRouteBeforePress).isA<ModalBottomSheetRoute<void>>();
+      final numPushedRoutesBeforePress = pushedRoutes.length;
+      await tester.tap(find.descendant(
+        of: find.byType(BottomSheet),
+        matching: buttonFinder));
+      await tester.pump(Duration.zero);
+
+      final newPushedRoute = pushedRoutes.skip(numPushedRoutesBeforePress)
+        .singleOrNull;
+
+      final sheetPopDuration = (topRouteBeforePress as ModalBottomSheetRoute<void>)
+        .reverseTransitionDuration;
+      // TODO not sure why a 1ms fudge is needed; investigate.
```

Comment:
Looks like the example provided upstream also has the same 1ms hack: https://github.com/flutter/flutter/pull/165832#issuecomment-2932867868.

Also the value of `sheetPopDuration` here turns out to be `200ms`, different from the previous `250ms`.

Reply:
I wonder if we tried 200ms when writing the test, but it failed for the same (unknown) reason it fails now, and we added 50ms to hack around instead of just 1ms.